primedev/core/math/vector.h

```diff
@@ -1,47 +1,331 @@
+#pragma once
+
+#include "bits.h"
+#include "math_pfns.h"
+
 #include <cmath>
 
-#pragma once
+#define DEG2RAD(a) (a) * (3.14159265358979323846f / 180.0f)
+#define RAD2DEG(a) (a) * (180.0f / 3.14159265358979323846f)
 
-union Vector3
+class Vector3
```

Comment:
I'm kinda surprised we're not using `Vector3` anywhere else so that changing it from `union` to a `class` would cause an issue ^^

Reply:
We are using it elsewhere, but not in any way that the struct doesn't support

Reply:
there was genuinely no point for this to exist ngl